src/drivers/barometer/bmp581/bmp581.cpp

```diff
@@ -0,0 +1,703 @@
+		* e.g. registers CHIP_ID, before the actual
+		* SPI communication with the device.
+		*/
+		status = _interface->get_reg(BMP5_REG_CHIP_ID_ADDR);
+	}
+
+	ret = _interface->set_reg(BMP5_SOFT_RESET_CMD, BMP5_REG_CMD_ADDR);
+
+	if(ret == OK) {
+		usleep(BMP5_DELAY_US_SOFT_RESET);
+
+		if(intf == BMP5_SPI_INTF) {
+			/* Performing a single read via SPI of registers,
+            		* e.g. registers CHIP_ID, before the actual
+            		* SPI communication with the device.
+            		*/
+			status = _interface->get_reg(BMP5_REG_CHIP_ID_ADDR);
+		}
+
+		status = _interface->get_reg(BMP5_REG_STATUS_ADDR);
+		/* Check if nvm_rdy status = 1 and nvm_err status = 0 to proceed */
+		if ((status & BMP5_INT_NVM_RDY) && (!(status & BMP5_INT_NVM_ERR))) {
+			status = _interface->get_reg(BMP5_REG_INT_STATUS_ADDR);
+
+			if(status & BMP5_INT_ASSERTED_POR_SOFTRESET_COMPLETE) {
+				result = true;
+			}
+		}
+	}
+
+	return result;
+}
+
+bool BMP581::set_config()
+{
+	bool result = false;
+	uint8_t rslt;
+	if(set_power_mode(BMP5_POWERMODE_STANDBY)) {
+
+		rslt = set_osr_odr_press_config();
+
+		if(rslt) {
+			rslt = set_iir_config();
+		}
+
+		if(rslt) {
+			rslt = configure_interrupt();
+		}
+
+		if(rslt) {
+			rslt = int_source_select();
+		}
+
+		if(rslt) {
+			result = true;
+		}
+	}
+
+	return result;
+}
+
+/*!
+ *  @brief This API is used to get interrupt status.
+ */
+bool BMP581::get_interrupt_status(uint8_t *int_status)
+{
+	bool result = false;
+	uint8_t status = 0xFF;
+
+	status = _interface->get_reg(BMP5_REG_INT_STATUS_ADDR);
+
+	result = true;
+	*int_status = status;
+
+	return result;
+}
+
+bool BMP581::check_deepstandby_mode(uint8_t *powermode)
+{
+	bool result = false;
+	int rslt;
+	uint8_t fifo_frame_sel;
+	uint8_t reg_data[2];
+
+	fifo_frame_sel = _interface->get_reg(BMP5_REG_FIFO_SEL_ADDR);
+	fifo_frame_sel = BMP5_GET_BITS_POS_0(fifo_frame_sel, BMP5_FIFO_FRAME_SEL);
+
+	rslt = _interface->get_reg_buf(BMP5_REG_OSR_CONFIG_ADDR, reg_data, 2);
+	uint8_t odr_reg = 0xFF;
+	if(rslt == OK) {
+		odr_reg = BMP5_GET_BITSLICE(reg_data[1], BMP5_ODR);
+	}
+
+	rslt = _interface->get_reg_buf(BMP5_REG_DSP_CONFIG_ADDR, reg_data, 2);
+	uint8_t set_iir_t_reg = 0xFF;
+	uint8_t set_iir_p_reg = 0xFF;
+	if(rslt == OK) {
+		set_iir_t_reg = BMP5_GET_BITS_POS_0(reg_data[1], BMP5_SET_IIR_TEMP);
+		set_iir_p_reg = BMP5_GET_BITSLICE(reg_data[1], BMP5_SET_IIR_PRESS);
+		result = true;
+	}
+
+   	/* As per datasheet odr should be less than 5Hz. But register value for 5Hz is less than 4Hz and so,
+     	* thus in this below condition odr is checked whether greater than 5Hz macro.
+     	*/
+	if((odr_reg > BMP5_ODR_05_HZ) && (fifo_frame_sel == BMP5_DISABLE) &&
+	   (set_iir_t_reg == BMP5_IIR_FILTER_BYPASS) && (set_iir_p_reg == BMP5_IIR_FILTER_BYPASS)) {
+		*powermode = BMP5_POWERMODE_DEEP_STANDBY;
+	}
+
+	return result;
+}
+
+/*!
+ *  @brief This API is used to get powermode of the sensor.
+ */
+bool BMP581::get_power_mode(uint8_t *power_mode)
```

Comment:
Rather than using a pointer and a bool return, add something like `POWERMODE_UNKNOWN` to indicate invalid status. Use the return early pattern to avoid all the nesting.